test/Core.ts

```diff
@@ -384,45 +386,110 @@ export class Foo {
         })
       })
 
-      it('should replace imports from the project with the local source folder (double quotes)', async () => {
-        const state = prefixWithCwd({
-          'package.json': defaultPackageJson,
-          'src/utils/foo.ts': `
+      describe('Package Imports', () => {
+        type PackageName = { scope: Option<string>; name: string }
+
+        type PackageImport = {
+          quotes: 'single' | 'double'
+          packageName: PackageName
+          lib: boolean
+          path: Option<string>
+        }
+
+        const stringifyPackageName = ({ scope, name }: PackageName): string =>
+          `${O.fold(
+            () => '',
+            (str) => `@${str}/`
+          )(scope)}${name}`
+
+        const stringifyPackageImport = ({ quotes, packageName, lib, path }: PackageImport) => {
+          const quote = quotes === 'single' ? "'" : '"'
+          const libStr = lib ? '/lib' : ''
+          const pathStr = O.fold(
+            () => '',
+            (str) => `/${str}`
+          )(path)
+          const packageNameStr = stringifyPackageName(packageName)
+          return `${quote}${packageNameStr}${libStr}${pathStr}${quote}`
+        }
+
+        const mkState = (packageName: PackageName, importLine: string) =>
+          prefixWithCwd({
+            'package.json': JSON.stringify({ name: stringifyPackageName(packageName), homepage: 'http://foo' }),
+            'src/utils/foo.ts': `
 /**
  * @since 0.0.1
  */
-
 /**
  * @category utils
  * @since 0.0.1
  */
 export class Foo {
   /**
    * @example
-   * import * as Foo from "${JSON.parse(defaultPackageJson)['name']}"
+   * ${importLine}
+   * 
+   * const bar = 1
    *
    * @since 0.0.1
    */
   public bar(): string {
     return 'bar'
   }
-}
-          `
-        })
-        const capabilities = makeCapabilities(state)
-
-        assertRight(await Core.main(capabilities)(), (result) => {
-          const exampleFileName = `${[...process.cwd().split('/'), 'src', 'utils', 'foo.ts'].join(
-            '-'
-          )}-Foo-method-bar-0.ts`
-          const exampleFilePath = `${process.cwd()}/docs/examples/${exampleFileName}`
-          const importFromRegex = new RegExp("from '../../src'")
+                            }`
+          })
+
+        /* By using the Array monad we generate a couple of PackageImport permutations
+         */
+        const packageImportPermutations: PackageImport[] = pipe(
+          A.bindTo('quotes')(['single' as const, 'double' as const]),
+          A.bind('scope', () => [O.some('my-org'), O.none]),
+          A.bind('name', () => ['my-package']),
+          A.bind('packageName', () =>
+            pipe(
+              A.bindTo('scope')([O.some('my-org'), O.none]),
+              A.bind('name', () => ['my-package'])
```

Comment:
This can be easily expanded to produce more tests. Maybe with `my_package` or `my_package0`...

Reply:
I like this for creating various permutations, but if it gets too unwieldy in the future we could consider using `fast-check` arbitraries which are probably more robust.